### ISSUE TITLE
Consider absorbing the ScalaSignature library from twitter/rsc

### ISSUE THREAD

**Issue by xeno-by** [scalameta/scalameta#1683]
Scalacp is currently based on Scalap, which has a number of known usability and correctness issues. https://github.com/twitter/rsc/pull/85 suggests an alternative by providing a library for read/write support for ScalaSignatures and both low-level and high-level representations for them. We could consider absorbing that library to improve Scalacp.

**Comment by xeno-by** [comment]
Here's the link to the API: https://github.com/twitter/rsc/tree/master/mjar/scalasig/src/main/scala/scala/meta/scalasig.

**Comment by xeno-by** [comment]
I don't think we should block the release of 4.0 for this. In fact, I'd prefer to leave this change to another release cycle, since it may break lots of stuff.

**Comment by tgodzik** [comment]
twitter/rsc is no longer being maintained, so I think it might no longer be relevant.
